web/packages/teleport/src/Assist/contexts/messages.tsx

```diff
@@ -253,14 +254,16 @@ async function convertServerMessage(
 
     const eventsResp = await api.fetch(sessionUrl + '/events');
     const sessionExists = eventsResp.status === 200;
-    const eventsData = (await eventsResp.json()) as {
-      events: SessionEvent[];
-    };
-    const execEvent = eventsData.events.find(isExecEvent);
 
     let msg;
     let errorMsg;
     if (sessionExists) {
+      // Get events only if the session exists. Otherwise, eventsData.events can be empty
+      // if the command execution failed.
+      const eventsData = (await eventsResp.json()) as {
+        events: SessionEvent[];
+      };
+      const execEvent = eventsData.events.find(isExecEvent);
```

Comment:
```suggestion
      // Get events only if the session exists. Otherwise, eventsData.events can be empty
      // if the command execution failed.
      const eventsData = (await eventsResp.json()) as {
        events: SessionEvent[];
      };
      const execEvent = eventsData.events?.find(isExecEvent);
```

we always assume that if we expect an array, the backend can return an [empty array as null](https://github.com/golang/go/issues/27589) unless the backend specifically defined it as a 0 length array (not sure if this endpoint will ever return a positive empty, but doesn't hurt to be safe)